File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function solution(n, k) {
    return n.toString(k).split('0').filter(x => x).reduce((acc, cur) => {
        if (cur < 2) return acc;
        for (let i = 2; i * i <= cur; i++)
            if (cur % i == 0) return acc;
        return acc + 1;
    }, 0);
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math
def converter(n, k):
    arr = []
    while n > 0:
        arr.append(str(n % k))
        n = n // k
    arr.reverse()
    return "".join(arr)

def isPrime(n):
    if (n < 2):
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if (n % i == 0):
            return False
    return True
def solution(n, k):
    answer = 0
    converted = converter(n, k)
    primes = converted.split('0')
    primes = list(filter(lambda x : x != "", primes))
    for p in primes:
        if (isPrime(int(p))):
            answer += 1
    return answer